src/Ext/TEvent/Body.cpp

```diff
@@ -159,6 +163,42 @@ bool TEventExt::VariableCheckBinary(TEventClass* pThis)
 	return false;
 }
 
+bool TEventExt::HousesAreDestroyedTEvent(TEventClass* pThis)
+{
+	int nIdxVariable = pThis->Value; //atoi(pThis->String);
+
+	if (nIdxVariable < 0)
+		return false;
+	
+	if (RulesExt::Global()->AIHousesLists.size() == 0)
+	{
+		Debug::Log("Map event %d: [AIHousesList] is empty. This event can't continue.\n", (int)pThis->EventKind);
+		return false;
+	}
+
+	std::vector<HouseTypeClass*> housesList = RulesExt::Global()->AIHousesLists.at(nIdxVariable);
+	
+	if (housesList.size() == 0)
+	{
+		Debug::Log("Map event %d: [AIHousesList](%d) is empty. This event can't continue.\n", (int)pThis->EventKind, nIdxVariable);
+		return false;
+	}
+
+	for (auto pTechno : *TechnoClass::Array)
+	{
+		if (ScriptExt::IsUnitAvailable(pTechno, false, true))
+		{
+			for (auto pHouse : housesList)
+			{
+				if (pTechno->Owner->Type == pHouse)
+					return false;
+			}
+		}
+	}
```

Comment:
```suggestion
    for (const auto pTechno : *TechnoClass::Array)
    {
		if (!ScriptExt::IsUnitAvailable(pTechno, false, true))
		    continue;
		for (const auto pHouse : housesList)
		{
			if (pTechno->Owner->Type == pHouse)
				return false;
		}
    }
```

Avoid nesting blocks if possible. Use `const` when possible.